components/renku_data_services/session/api.spec.yaml

```
@@ -377,16 +385,23 @@ components:
         is_archived:
           $ref: "#/components/schemas/IsArchived"
           default: false
+        environment_image_source:
+          $ref: "#/components/schemas/EnvironmentImageSourceImage"
       required:
         - name
         - container_image
+        - environment_image_source
     EnvironmentPatchInLauncher:
       allOf:
-        - $ref: "#/components/schemas/EnvironmentPatch"
+        - oneOf:
+          - $ref: "#/components/schemas/EnvironmentPatch"
+          - $ref: "#/components/schemas/BuildParametersPatch"
         - type: object
           properties:
-            environment_kind:
-              $ref: "#/components/schemas/EnvironmentKind"
+          environment_kind:
+            $ref: "#/components/schemas/EnvironmentKind"
```

Schema does not validate:

```suggestion
            environment_kind:
              $ref: "#/components/schemas/EnvironmentKind"
```